timur/lib/client/jsx/components/messages.jsx

```diff
@@ -42,39 +47,92 @@ class Messages extends React.Component{
       nav = <div id="nav">
         {
-          current_message ?  <span className="arrow fas fa-chevron-left" onClick={ this.prevMessage.bind(this) }/> : null
+          current_message ? <span className="arrow fas fa-chevron-left" onClick={this.prevMessage.bind(this)}/> : null
         }
         <span className="pager">
-          Page { current_message + 1 } of { messages.length }
+          Page {current_message + 1} of {messages.length}
         </span>
         {
-          current_message < messages.length-1 ?  <span className="arrow fas fa-chevron-right" onClick={ this.nextMessage.bind(this) }/> : null
+          current_message < messages.length - 1 ?
+            <span className="arrow fas fa-chevron-right" onClick={this.nextMessage.bind(this)}/> : null
         }
       </div>
     }
 
-    let message = markdown( sanitize( messages[current_message] ) );
-
-    return <div id="messages">
-            <div id="quote">
-              <svg width="31" height="40">
-                <path d="M 2,40 20,25 18,40"/>
-              </svg>
-            </div>
-            <div id="dismiss" onClick={ this.dismissMessages.bind(this) }>
-              <span className="fas fa-check"> </span>
-            </div>
-            <div id="message_viewer">
-              { nav }
-              <div id="message"
-                dangerouslySetInnerHTML={ { __html: message } }/>
-            </div>
-          </div>;
+    let message = markdown(sanitize(messages[current_message]));
+
+    return <MessagesToNotificationsBridge messages={messages}>
+      <div id="messages">
+        <div id="quote">
+          <svg width="31" height="40">
+            <path d="M 2,40 20,25 18,40"/>
+          </svg>
+        </div>
+        <div id="dismiss" onClick={this.dismissMessages.bind(this)}>
+          <span className="fas fa-check"> </span>
+        </div>
+        <div id="message_viewer">
+          {nav}
+          <div id="message"
+               dangerouslySetInnerHTML={{ __html: message }}/>
+        </div>
+      </div>
+    </MessagesToNotificationsBridge>;
   }
 }
 
+// Used to enable feature flag backed notifications override for existing messages actions.
+// If this goes well, we should remove this component and replace showMessages invocations.
+export function MessagesToNotificationsBridge({ children, messages }) {
+  const { addLocalNotification, removeAllLocalNotifications } = useNotifications();
+  if (!useFeatureFlag('newNotifications')) return children;
+
+  useEffect(() => {
+    removeAllLocalNotifications();
+
+    messages.forEach((message, i) => {
+      // let content = markdown(sanitize(message));
+      addLocalNotification(i + '', {
+        // content: <div id="message" dangerouslySetInnerHTML={{ __html: content }}/>,
+        ...unMarkdownNotification(message),
+        container: 'top-center',
+        type: 'warning',
+      })
+    })
+  }, [messages])
+
+  return null;
+}
+
+export function unMarkdownNotification(md) {
```

Comment:
Might eventually be worth having server pass down notifications / errors that do not require markdown.  From what I can tell, the markdown usage basically boils down to

1.  A title with `##`
2.  Maybe a list with `*`
3.  Bold or italics.

An alternative might be to just include these components as data elements of a json payload, than requiring the client to parse markdown.

That said, this bridge here works just fine for now.